Docs(Next-Gen): add NG Careamics docs (#804)
CAREamics/careamics: docs/careamist_predicting.md

```python title="Prediction"
--8<-- "careamist_predicting.py:pred_tiled"
```

1. The `tile_size` need not be equal to the training patch size.
2. Overlaps are optional, and default to `(48, 48)`

!!! note "Why use tiling?"

    If the images have dimensions that are not comaptible with the network architecture
    (`shape % 2**model_depth != 0`), then tiling is required to ensure that the network
    can process the images.

    If the images are large, then they might simply not fit in memory and need to be
    processed as tiles.

!!! note "What tiling size to choose?"

    The tiling size should be chosen based on the size of the images and the available
    memory. A good starting point is to use a patch size that is a multiple of
    `2**model_depth`. You can then play with `batch_size` to find the optimal memory
    usage.

    Note that the overlaps must be larger than the receptive field of the network.

!!! warning "Dimensions"

    Obviously, the tiling and overlaps must respect the dimensions of the images (2D
    or 3D).

### Changing dataloading parameters

During prediction, you can change the dataloading parameters by passing them to the

> `predict` method. Currently, we only expose `batch_size` and `num_workers`.

Reviewer: Not sure if this line would be clear, maybe
```suggestion
`predict` method. The parameters `batch_size` and `num_workers` can be set through the `predict` function arguments, while any other parameters have to be changed through the configuration.
```

And btw I was just looking through the configuration factories and `pred_dataloader_params` is not available through `create_advanced_n2v_config`, don't know if we want to add that

Author: Either add them or add a parameter to the prediction functions yes.

```python title="Prediction"
--8<-- "careamist_predicting.py:pred_dataloader"
```

### Data parameters

The data we want to predict on might be different from the training data, in terms of
axes or format. The `predict` method allows changing these parameters (`axes`, 
`data_type`, `channels`, `in_memory`).

Here, let's say we trained from arrays of axes `YX`, and now want to predict with the
trained model on a TIFF file (we have a path `pred_data_path`) that has channels. Since

> we did not trained on channels, we need to pass `channels` to specify a single channel

Reviewer: typo
```suggestion
we did not train on channels, we need to pass `channels` to specify a single channel
```

> we did not trained on channels, we need to pass `channels` to specify a single channel

Reviewer: Maybe having a file with `S` samples is a more common situation to have as an example?

Author: Done in https://github.com/CAREamics/careamics/pull/804/changes/6739c2483b100eba096bce361de0d6c175fd5de5

> we want to train on. We also need to specify the new axes and data_type. Finally, we

Reviewer: ```suggestion
we want to train on. We also need to specify the new axes and `data_type`. Finally, we
```
